mausignald/rpc.py

```diff
@@ -59,29 +63,37 @@ async def wait_for_connected(self, timeout: Optional[int] = None) -> bool:
     async def connect(self) -> None:
         if self._writer is not None:
             return
-
+        self.log.info("Connecting to signald")
         self._communicate_task = asyncio.create_task(self._communicate_forever())
         await self._connect_future
 
     async def _communicate_forever(self) -> None:
-        while True:
-            try:
-                self._reader, self._writer = await asyncio.open_unix_connection(
-                    self.socket_path, limit=_SOCKET_LIMIT)
-            except OSError as e:
-                self.log.error(f"Connection to {self.socket_path} failed: {e}")
-                await asyncio.sleep(5)
-                continue
-
-            read_loop = asyncio.create_task(self._try_read_loop())
-            self.is_connected = True
-            await self._run_rpc_handler(CONNECT_EVENT, {})
-            self._connect_future.set_result(True)
-
-            await read_loop
-            self.is_connected = False
-            await self._run_rpc_handler(DISCONNECT_EVENT, {})
-            self._connect_future = self.loop.create_future()
+        try:
+            while True:
+                    try:
+                        self._reader, self._writer = await asyncio.open_unix_connection(
+                            self.socket_path, limit=_SOCKET_LIMIT)
+                    except OSError as e:
+                        self.log.error(f"Connection to {self.socket_path} failed: {e}, retrying in 5s")
+                        await asyncio.sleep(5)
+                        continue
+
+                    self.log.debug(f"Connection to {self.socket_path} succeeded")
+                    read_loop = asyncio.create_task(self._try_read_loop())
+                    self.is_connected = True
+                    CONNECTED_GAUGE.set(1)
+                    await self._run_rpc_handler(CONNECT_EVENT, {})
+                    self._connect_future.set_result(True)
+
+                    await read_loop
+                    self.is_connected = False
+                    CONNECTED_GAUGE.set(0)
+                    await self._run_rpc_handler(DISCONNECT_EVENT, {})
+                    self._connect_future = self.loop.create_future()
+                    RECONNECTIONS_COUNTER.inc(1)
```

Comment:
This block seems over-indented

Reply:
Have split out the loop from the code, i'd still like the try catch to just log when we're going to have a fatal error, because otherwise this failed silently (when we have something catastrophic)

Reply:
I meant it was indented with 4 spaces more than needed, but that works too

Reply:
Oh, I thought you reasonably objected to massive try catch blocks :D. well, hopefully it's less